Source/ARTInternalLog.m

```diff
@@ -1,19 +1,25 @@
 #import "ARTInternalLog.h"
+#import "ARTInternalLog+Testing.h"
 #import "ARTInternalLogCore.h"
 #import "ARTVersion2Log.h"
-
-NS_ASSUME_NONNULL_BEGIN
-
-@interface ARTInternalLog ()
-
-@property (nonatomic, readonly) id<ARTInternalLogCore> core;
-
-@end
-
-NS_ASSUME_NONNULL_END
+#import "ARTLogAdapter.h"
 
 @implementation ARTInternalLog
 
++ (ARTInternalLog *)classMethodLogger {
```

Comment:
Usually it's "shared" or "default", hence `sharedLogger` or `defaultLogger`.

Reply:
Could it be `sharedLogger_onlyForUseInClassMethods`? I want it to be very obvious when we're misusing it (since the number of situations in which we're meant to use it is tiny).

Reply:
Maybe to add a comment that states that it should be used only in class methods?

Reply:
It is already documented, explaining the limited situations in which it's meant to be used.

I've renamed it to `sharedClassMethodLogger_readDocumentationBeforeUsing`. Given we use it in only two places, I think the long name is fine.